src/main/scala/mesosphere/marathon/core/health/HealthCheck.scala

```diff
@@ -10,146 +9,210 @@ import org.apache.mesos.{ Protos => MesosProtos }
 
 import scala.concurrent.duration._
 
-case class HealthCheck(
+sealed trait HealthCheck extends MarathonState[Protos.HealthCheckDefinition, HealthCheck] {
+  def gracePeriod: FiniteDuration
+  def interval: FiniteDuration
+  def timeout: FiniteDuration
+  def maxConsecutiveFailures: Int
+  def toProto: Protos.HealthCheckDefinition
 
-  path: Option[String] = HealthCheck.DefaultPath,
+  override def version: Timestamp = Timestamp.zero
+  def mergeFromProto(bytes: Array[Byte]): HealthCheck =
+    mergeFromProto(Protos.HealthCheckDefinition.parseFrom(bytes))
+  def mergeFromProto(proto: Protos.HealthCheckDefinition): HealthCheck = HealthCheck.mergeFromProto(proto)
 
-  protocol: Protocol = HealthCheck.DefaultProtocol,
+  protected def protoBuilder: Protos.HealthCheckDefinition.Builder =
+    Protos.HealthCheckDefinition.newBuilder
+      .setGracePeriodSeconds(this.gracePeriod.toSeconds.toInt)
+      .setIntervalSeconds(this.interval.toSeconds.toInt)
+      .setTimeoutSeconds(this.timeout.toSeconds.toInt)
+      .setMaxConsecutiveFailures(this.maxConsecutiveFailures)
+}
 
-  portIndex: Option[Int] = HealthCheck.DefaultPortIndex,
+trait HealthCheckWithPort {
```

Comment:
Both fields are exclusive, and there is a validation that makes sure that at least one of them is set.